ngraph_creator/operations/src/FullyConnected.cpp

```diff
@@ -0,0 +1,94 @@
+#include <FullyConnected.hpp>
+
+namespace android {
+namespace hardware {
+namespace neuralnetworks {
+namespace nnhal {
+
+FullyConnected::FullyConnected(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+// Supports only FP32 input. Will add support for QUANT8 through decompose node
+// once the vpu and gna plugin support if confirmed
+bool FullyConnected::validate() {
+    auto input0 = getInputOperand(0);
+    auto input2 = getInputOperand(2);
+    auto output0 = getOutputOperand(0);
+
+    if (input0.type != OperandType::TENSOR_FLOAT32) {
```

Comment:
Can you not use checkInputOperandType and checkOutputOperandType. So, you would not need the getInputOperand too.

Reply:
Validation on operand 0 happens multiple times within the validate function. Also the verification of operand type for operands 0 and 2. Instead of accessing the vector of operations every time we need to perform a check on operand, a single access to the operand data type is better?